napalm_eos/eos.py

```diff
@@ -1201,3 +1201,162 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
                 previous_probe_ip_address = ip_address
 
         return traceroute_result
+
+    def get_bgp_neighbors_detail(self, neighbor_address=''):
+        """Function to return detailed BGP information.
+
+        Information returned by this function can be about a single peer or
+        about all peers. This can be controlled using the following:
+
+        :params neighbor_address: Retuns the statistics for a specific
+                                  BGP neighbor or for all BGP neighbors.
+        """
+
+        peer_ver = None
+        is_all = True if not neighbor_address else False
+
+        if is_all:
+            command = ['show ip bgp neighbors vrf all']
+        else:
+            peer_ver = IPAddress(neighbor_address).version
```

Comment:
I usually prefer to catch this kind of exceptions (e.g.: https://github.com/napalm-automation/napalm-nxos/blob/develop/napalm_nxos/nxos.py#L694) since they won't provide any useful input, but leaving potential vendor exception uncaught (e.g.: datatype of a field changed between firmwares etc) that provides more useful feedback.